tests/conftest.py

```diff
@@ -1,8 +1,30 @@
+import asyncio
+import sys
+
+if sys.version_info >= (3, 7):
+    from contextlib import asynccontextmanager
+else:
+    from async_generator import asynccontextmanager
```

Comment:
From experience, a great approach for this kind of thing is to have a dedicated "compatibility" module.

Should we move this to a new `tests/compat.py`?

Also I think we could just try/except:

```suggestion
try:
    from contextlib import asynccontextmanager
except ImportError:  # pragma: no cover
    from async_generator import asynccontextmanager
```